Repo: typecheck failures on `main`: website's `data/recent-blog-posts.json` not created

### Suggestion

Example from https://github.com/typescript-eslint/typescript-eslint/actions/runs/21600477673/job/62257242979:

```plaintext
❌ > nx run website:typecheck
  
  > tsc --build --emitDeclarationOnly
  
  Error: src/components/home/RecentBlogPosts/index.tsx(2,25): error TS2307: Cannot find module '@site/data/recent-blog-posts.json' or its corresponding type declarations.
  Error: src/components/home/RecentBlogPosts/index.tsx(18,28): error TS7006: Parameter 'post' implicitly has an 'any' type.
  > tsc --build --emitDeclarationOnly
  
  Error: src/components/home/RecentBlogPosts/index.tsx(2,25): error TS2307: Cannot find module '@site/data/recent-blog-posts.json' or its corresponding type declarations.
  Error: src/components/home/RecentBlogPosts/index.tsx(18,28): error TS7006: Parameter 'post' implicitly has an 'any' type.
  Warning: command "tsc --build --emitDeclarationOnly" exited with non-zero status code::endgroup::
```

### Additional Info

The website's `data/recent-blog-posts.json` is created by a Docusaurus plugin in build. I'm thinking that that's not being cached as an Nx output, and so isn't available during `typecheck` after `build`?

💖

--- a/packages/website/package.json
+++ b/packages/website/package.json
@@ -116,7 +116,8 @@
         },
         "cache": true,
         "outputs": [
-          "{projectRoot}/build"
+          "{projectRoot}/build",
+          "{projectRoot}/data"
         ]
       },
       "lint": {

